feature/amazon/src/main/java/com/revenuecat/purchases/amazon/AmazonBilling.kt

```diff
@@ -90,6 +92,39 @@ internal class AmazonBilling constructor(
         )
     }
 
+    override fun normalizePurchaseData(
+        productID: String,
+        purchaseToken: String,
+        storeUserID: String,
+        onSuccess: (correctProductID: String) -> Unit,
+        onError: (PurchasesError) -> Unit
+    ) {
+        val currentlyCachedTokensToSkus = cache.getReceiptSkus()
+
+        currentlyCachedTokensToSkus[purchaseToken]?.let { sku ->
+            onSuccess(sku)
+            return
+        }
+
+        amazonBackend.getAmazonReceiptData(
+            purchaseToken,
+            storeUserID,
+            onSuccess = { response ->
+                log(LogIntent.DEBUG, AmazonStrings.RECEIPT_DATA_RECEIVED.format(response.toString()))
+
+                if (response.has(TERM_SKU_JSON_KEY)) {
+                    val termSku = response[TERM_SKU_JSON_KEY] as String
```

Comment:
do we need a try/catch on this?

Reply:
Hmmm, good point. I am not sure why Android Studio is not suggesting it since get can throw `JSONException`